vivisect/qt/main.py

```diff
@@ -201,6 +202,113 @@ def addBookmark(self, va, parent=None):
         if ok:
             self.vw.setVaSetRow('Bookmarks', (va, str(bname)))
 
+    def getMemoryWidgets(self):
+        return self.views.get('VQVivMemoryView')
+
+    def getMemWidgetsByName(self, name='viv', firstonly=True):
+        '''
+        Returns a list of Memory View Widgets with the given name.
+        If "firstonly" is True, only return the first one or None(not a list)
+
+        Returns a tuple of (Widget, DockWidget).  The "Widget" is obtained from
+        the DockWidget, but they both have different powers.
+        '''
+        logger.debug("getWindowsByName(%r, firstonly=%r)", name, firstonly)
+        out = []
+
+        for vqDW in self.getMemoryWidgets():
+            w = vqDW.widget()
+            logger.debug("enn: " + w.getEnviNavName())
+            if w.getEnviNavName() == name:
+                if firstonly:
+                    return w, vqDW
+
+                out.append((w,vqDW))
+
+        if firstonly:   # if firstonly and we don't have one, return None
+            return None
+
+        return out
+
+    def getFuncGraphs(self):
+        return self.views.get('VQVivFuncgraphView')
+
+    def getFuncGraphsByName(self, name='FuncGraph0', firstonly=True):
+        '''
+        Returns a list of Dock Widgets which have a "getEnviNavName"
+        This includes MemoryViews and FuncGraphs
+        '''
+        logger.debug("getFuncGraphsByName()")
+
+        out = []
+        for vqDW in self.getFuncGraphs():
+            w = vqDW.widget()
+            logger.debug("enn: " + w.getEnviNavName())
```

Comment:
Again, `enn:`?

Reply:
Also, I dislike how extremely similar these two functions are. I'm fairly certain we could merge them.

Reply:
killed enn logs (EnviNavName).  it was a little too "devvy"

Reply:
i'm open to suggestions... i've considered merging them, but not sure how best to do it from a usability standpoint.